pkg/apis/cloudcredential/v1/openstack_types.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2019 The OpenShift Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TODO: these types should eventually be broken out, along with the actuator,
+// to a separate repo.
+
+// OpenStackProviderSpec the specification of the credentials request in OpenStack.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type OpenStackProviderSpec struct {
+	metav1.TypeMeta `json:",inline"`
+}
```

Comment:
The empty spec here is interesting. For AWS we did two modes, minting where we create the creds you specifically asked for, and passthrough where if we don't have enough permission to mint we just copy the root creds everywhere. 

With the empty spec here and the pure copying this is similar to what MS did for Azure's initial cut which is just passthrough. There was an additional bit of functionality where we check and annotate the root creds secret with which mode they support which you can see here: https://github.com/openshift/cloud-credential-operator/tree/master/pkg/controller/secretannotator. Ideally it would be good to do the same for OpenStack even though we're just always saying passthrough, we'd be a little closer to a full implementation.

Do we plan to fill this out with the ability to mint fine grained credentials/permissions in the future?

Reply:
Hi! Yes, absolutely we have these plans. I decided to keep things simple at first and just copy the data from the system secret, because it allows us to easily test our patches for operators.
When the codebase is stable enough I'm going to implement minting of fine grained credentials, like it's done with Azure, i.e. with Rolebindings in the spec.

Reply:
Excellent, go for AWS as an example when you get to that point, Azure doesn't have full minting yet they're just passthrough as well so far.